src/game_display.cpp

```diff
@@ -664,3 +668,97 @@ display::overlay_map& game_display::get_overlays()
 {
 	return overlay_map_;
 }
+
+std::vector<texture> game_display::get_reachmap_images(const map_location& loc)
+{
+	// Use get_direction() from display.cpp namespace display_direction
+	using namespace display_direction;
+	std::vector<std::string> names;
+	const auto adjacent = get_adjacent_tiles(loc);
+
+	enum visibility { REACH = 0, CLEAR = 1 };
+	std::array<visibility, 6> tiles;
+
+	const std::string* image_prefix_ = &game_config::reach_map_prefix;
+	DBG_DP << "Loaded image prefix: " << game_config::reach_map_prefix;
+
+	for(int i = 0; i < 6; ++i) {
+		if(reach_map_.find(adjacent[i]) != reach_map_.end()) {
+			DBG_DP << "Adjacent " << std::to_string(i) << " to tile " << loc << " is REACHABLE";
+			tiles[i] = REACH;
+		} else {
+			DBG_DP << "Adjacent " << std::to_string(i) << " to tile " << loc << " is NOT REACHABLE";
+			tiles[i] = CLEAR;
+		}
+	}
+
+	for(int v = REACH; v != CLEAR; ++v) {
```

Comment:
This is a for loop just so you can use continue? Seems like that could be rewritten more clearly.

Reply:
You're right. This is translatable to saying:
`for(int v = 0; v != 1; ++v) {`
Which serves no purpose. variable "v" could also be dropped then.

Reply:
Hopefully solved with [195105e](https://github.com/wesnoth/wesnoth/pull/9292/commits/195105e4920f1c68f465221f7aef6a8e74df0745)